extensions/amp-selector/1.0/amp-selector.js

```diff
@@ -249,12 +250,42 @@ function OptionShim({
 function SelectorShim({
   shimDomElement,
   children,
+  form,
   multiple,
+  name,
   disabled,
   onKeyDown,
   role = 'listbox',
   tabIndex,
+  value,
 }) {
+  const input = useRef(
```

Comment:
This is certainly ok to keep this way. But could you pls also lookup in the background https://html.spec.whatwg.org/multipage/custom-elements.html#dom-elementinternals-setformvalue and we can chat about this offline. I think this is future way to do the same but avoiding a hidden input. This may or may not apply to the Preact layer.